### PR TITLE
Write main test for node

### PR DESCRIPTION
**Pull request by priyawadhwa**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -42,6 +42,7 @@ script:
  # - bazel clean && bazel build //...
  - bazel test reconciletags:reconciletags_test
  - bazel run //ftl:node_builder_test
```
Comment (dlorenc) [inline .travis.yml]: I don't think this one needs a "bazel run", could you combine these two to "bazel test //ftl:..."

File: ftl/node/main_test.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+                                       ) as img:
+            self.registry.setImage('fake.gcr.io/base/image:appended', img)
+
+        self.AssertPushed(self.registry, 'fake.gcr.io/base/image:initial')
+        self.AssertPushed(self.registry, 'fake.gcr.io/base/image:appended')
+
+        append_layer_mock.return_value = self.registry.getImage(
+                                         'fake.gcr.io/base/image:appended')
+
+        sys.argv = ["main.py",  "--base=fake.gcr.io/base/image:initial",
```
Comment (dlorenc) [inline ftl/node/main_test.py]: Could you make a slight refactor to main to take these as parameters? Then you can pass it in instead of setting sys.argv.

**Comment by aaron-prindle** [comment]
Is main_test.py going going to house all of the integration tests for ftl node?  What is the primary functionality intended to be tested for main.py in main_test.py?

**Comment by priyawadhwa** [comment]
Right now main_test.py primarily tests that the image is uploaded correctly, but I'm not sure if it's where all of the integration tests are going to be...

File: .travis.yml [hunk 3]
```diff
@@ -41,7 +41,7 @@ script:
  - ./.buildifier.sh
  # - bazel clean && bazel build //...
  - bazel test reconciletags:reconciletags_test
- - bazel run //ftl:node_builder_test
+ - bazel test ftl:node_builder_test ftl:node_main_test
  - bazel test testing/lib:mock_registry_tests
```
Comment (dlorenc) [inline .travis.yml]: could you make this :... to test all the files in this package?

File: ftl/node/main.py [hunk 4]
```diff
@@ -47,9 +47,16 @@
     help='The path where the application data sits.')
 
 
-def main():
+def main(base="", name="", directory=""):
```
Comment (dlorenc) [inline ftl/node/main.py]: Ah sorry, I wasn't very clear here. Could you pass sys.argv to main from the bottom of this file, and pass that parameter to parser.parse_args?

File: ftl/node/main.py [hunk 5]
```diff
@@ -90,4 +91,4 @@ def main():
 
 
 if __name__ == '__main__':
-    main()
+    main(sys.argv)
```
Comment (dlorenc) [inline ftl/node/main.py]: One more tweak, this is a bit easier to use if you pass argv[1:] here, then you don't need the "main.py" in your test.

Comment (priyawadhwa) [reply ftl/node/main.py]: This all makes a lot more sense, thanks Dan!

**Pull request merged**